File: blog-server-java/src/main/java/com/xiong/utils/Result.java
package com.xiong.utils;

import org.springframework.stereotype.Component;

@Component
public class Result {
    private int code;
    private Object data;
    private String msg;


    public Result() {
    }

    public Result(Object d) {
        this.data = d;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public void setData(Object data) {
        this.data = data;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public int getCode() {
        return code;
    }

    public Object getData() {
        return data;
    }

    public String getMsg() {
        return msg;
    }
}

File: blog-server-ts/src/utils/result.ts
import { ObjType } from "./types";

export class Result<T = null> {
  private code: number;
  private data: T;
  private message: string;
  private success: boolean;

  constructor() {
    this.code = 200;
    this.success = true;
    this.data = null as T;
    this.message = "";
  }
  public getCode(): number {
    return this.code;
  }
  public getSuccess(): boolean {
    return this.success;
  }
  public getData(): T {
    return this.data;
  }
  public getMessage(): string {
    return this.message;
  }
  public setCode(code: number): void {
    this.code = code;
  }
  public setSuccess(success: boolean): void {
    this.success = success;
  }
  public setData(data: T) {
    this.data = data;
  }
  public setMessage(message: string): void {
    this.message = message;
  }
}

interface ListDataType<T = ObjType> {
  totalCount: number;
  list: T[];
}

export class ListResult<T> extends Result<ListDataType<T>> {}
